pkg/controllers/util.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"fmt"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	jobset "sigs.k8s.io/jobset/api/v1alpha1"
+)
+
+func IsJobFinished(job *batchv1.Job) (bool, batchv1.JobConditionType) {
```

Comment:
It is used in the integration tests, which are in a different package under the `/test/integration` directory and not included in this PR (trying to keep the scope of this PR smaller/focused since it is already quite large).